contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -1588,6 +1590,220 @@ create_xp_instance_regread_in_master_dbo_internal(PG_FUNCTION_ARGS)
 	PG_RETURN_INT32(0);
 }
 
+Datum
+sp_execute_postgresql(PG_FUNCTION_ARGS)
+{
+	List	   *parsetree_list;
+	char	   *postgresStmt;
+	Node	   *stmt;
+	Node	   *parsetree;
+	size_t		len;
+	PlannedStmt *wrapper;
+	bool flag = false;
+	const char *allowed_extns[] = {"pg_stat_statements", "tds_fdw", "fuzzystrmatch"};
+	int allowed_extns_size = sizeof(allowed_extns) / sizeof(allowed_extns[0]);
+	const char *saved_dialect = GetConfigOption("babelfishpg_tsql.sql_dialect", true, true);
+	Oid			current_user_id = GetUserId();
+	const char *saved_path = pstrdup(GetConfigOption("search_path", true, true));
+	const char *new_path = "public, \"$user\", sys, pg_catalog";
+	
+	PG_TRY();
+	{
+		set_config_option("babelfishpg_tsql.sql_dialect", "postgres",
+						GUC_CONTEXT_CONFIG,
+						PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);
+		
+		postgresStmt = PG_ARGISNULL(0) ? NULL : TextDatumGetCString(PG_GETARG_TEXT_PP(0));
+
+		if (postgresStmt == NULL)
+				ereport(ERROR, (errcode(ERRCODE_NULL_VALUE_NOT_ALLOWED),
+								errmsg("statement cannot be NULL")));
+
+		/* Remove trailing whitespaces */
+		len = strlen(postgresStmt);
+		while (isspace(postgresStmt[len - 1]))
+			postgresStmt[--len] = 0;
+
+		/* check if input statement is empty after removing trailing spaces */
+		if (len == 0)
+			ereport(ERROR, (errcode(ERRCODE_NULL_VALUE_NOT_ALLOWED),
+							errmsg("statement cannot be NULL")));
+
+		parsetree_list = raw_parser(postgresStmt, RAW_PARSE_DEFAULT);
+		stmt = parsetree_nth_stmt(parsetree_list, 0);
+		rewrite_object_refs(stmt);
```

Comment:
Please remove this function call. It TSQL specific rewrite. We don't need it here.